serialization/TransactionOutput/TransactionOutput.go

```diff
@@ -194,11 +195,28 @@ func (to *TransactionOutput) GetDatumHash() *serialization.DatumHash {
 
 func (to *TransactionOutput) GetDatum() *PlutusData.PlutusData {
 	if to.IsPostAlonzo {
-		return to.PostAlonzo.Datum
+		switch d := to.PostAlonzo.Datum; d.DatumType {
+		case PlutusData.DatumTypeHash:
+			return nil
```

Comment:
here we return nil, but below we return `&PlutusData.PlutusData{}`; is there a meaningful distinction between these?

Reply:
https://github.com/SundaeSwap-finance/apollo/blob/b16720edb181e149450146314a0b79f30a05b097/serialization/PlutusData/PlutusData.go#L420-L428
A PlutusData with a zero PlutusDataType field is the representation of an empty array, which I think is technically wrong here.